boards/metro_m4/examples/qspi.rs

```diff
@@ -0,0 +1,43 @@
+#![no_std]
+#![no_main]
+
+extern crate cortex_m;
+extern crate cortex_m_semihosting;
+extern crate metro_m4 as hal;
+#[cfg(not(feature = "use_semihosting"))]
+extern crate panic_halt;
+#[cfg(feature = "use_semihosting")]
+extern crate panic_semihosting;
+
+use hal::clock::GenericClockController;
+use hal::prelude::*;
+use hal::entry;
+use hal::pac::{CorePeripherals, Peripherals};
+use hal::qspi::Qspi;
+
+#[entry]
+fn main() -> ! {
+    let mut peripherals = Peripherals::take().unwrap();
+    let core = CorePeripherals::take().unwrap();
+    let clocks = GenericClockController::with_external_32kosc(
+        peripherals.GCLK,
+        &mut peripherals.MCLK,
+        &mut peripherals.OSC32KCTRL,
+        &mut peripherals.OSCCTRL,
+        &mut peripherals.NVMCTRL,
+    );
+    let mut pins = hal::Pins::new(peripherals.PORT);
+
+    let qspi = Qspi::new(
+        &mut peripherals.MCLK, 
+        &mut pins.port,
+        peripherals.QSPI, 
+        pins.flash_sck,
+        pins.flash_cs,
+        pins.flash_io0,
+        pins.flash_io1,
+        pins.flash_io2,
+        pins.flash_io3,
+    );
+    loop{}
```

Comment:
Nit: Could you make the example do something (maybe flash an LED with the results from reading the first byte) to demonstrate more than initialization?

Reply:
I made the example erase the chip, write something, read it back, and make the led solid if it matches, blinky if it doesn't. It blinked :'(